### PR TITLE
xds: Fix bug of locality store not update weights for existing localities

### PR DESCRIPTION
**Pull request by voidzcy**
As discussed in https://github.com/grpc/grpc-java/pull/5973#discussion_r302289849, this PR fixed the bug and added test coverage without modifying `WeightChildPicker` implementation.

### CODE REVIEW

File: xds/src/test/java/io/grpc/xds/LocalityStoreTest.java [hunk 1]
```diff
@@ -456,22 +475,27 @@ public PickResult pickSubchannel(PickSubchannelArgs args) {
...
-    ArgumentCaptor<SubchannelPicker> subchannelPickerCaptor31 =
-        ArgumentCaptor.forClass(SubchannelPicker.class);
-    verify(helper).updateBalancingState(same(READY), subchannelPickerCaptor31.capture());
+    ArgumentCaptor<SubchannelPicker> subchannelPickerCaptor =
+        ArgumentCaptor.forClass(null);
+    verify(helper).updateBalancingState(same(READY), subchannelPickerCaptor.capture());
     assertThat(pickerFactory.totalReadyLocalities).isEqualTo(1);
-    assertThat(
-            subchannelPickerCaptor31.getValue().pickSubchannel(pickSubchannelArgs).getSubchannel())
+    pickerFactory.nextIndex = 0;
```
Comment (zhangkun83) [inline xds/src/test/java/io/grpc/xds/LocalityStoreTest.java]: Why is this line necessary?

Comment (voidzcy) [reply xds/src/test/java/io/grpc/xds/LocalityStoreTest.java]: Here we have only one `WeightChildPicker` in list (the previous line verified that). So setting the index to 0 explicitly is really to ensure nothing unexpected would happen. `pickerFactory.nextIndex` is 0 by default, but we shouldn't rely on that.
